pinot-common/src/main/java/com/linkedin/pinot/common/utils/request/RequestUtils.java

```diff
@@ -206,4 +212,43 @@ public static boolean isFitForStarTreeIndex(SegmentMetadata segmentMetadata, Fil
 
     return true;
   }
+
+  /**
+   * Must be called after a successful check of {@link #isFitForStarTreeIndex}
+   * @return true if no fasthll aggregation or rewriting successful, false otherwise
+   */
+  public static boolean performFastHllRewriting(SegmentMetadata segmentMetadata, BrokerRequest brokerRequest) {
+    List<AggregationInfo> aggregationsInfo = brokerRequest.getAggregationsInfo();
+    StarTreeMetadata starTreeMetadata = segmentMetadata.getStarTreeMetadata();
+    Map<AggregationInfo, String> fastHllAggregationInfoToDerivedColumnNameMap = new HashMap<>();
+    // Check if all fasthll derived columns exist
+    for (AggregationInfo aggregationInfo : aggregationsInfo) {
+      if (aggregationInfo.getAggregationType().toLowerCase().equals("fasthll")) {
+        String derivedColumn = extractDerivedColumn(aggregationInfo, starTreeMetadata);
```

Comment:
we should also support case where hll column is provided by customer and not generated by us. Let's discuss this part. You can support that in next PR